packages/protocol/contracts/governance/Validators.sol

```diff
@@ -524,21 +434,29 @@ contract Validators is
   }
 
   /**
-   * @notice De-registers a validator, removing it from the group for which it is a member.
-   * @param index The index of this validator in the list of all validators.
+   * @notice De-registers a validator.
+   * @param index The index of this validator in the list of all registered validators.
    * @return True upon success.
    * @dev Fails if the account is not a validator.
    */
   function deregisterValidator(uint256 index) external nonReentrant returns (bool) {
     address account = getLockedGold().getAccountFromActiveValidator(msg.sender);
     require(isValidator(account));
+
+    // Require that the validator has not been a member of a validator group for
+    // `validatorLockedGoldRequirements.duration` seconds.
     Validator storage validator = validators[account];
     if (validator.affiliation != address(0)) {
-      _deaffiliate(validator, account);
+      require(!groups[validator.affiliation].members.contains(account));
```

Comment:
The validator needs to be removed from the group's membership list before `deregisterValidator` is called. Furthermore, it needs to have been removed from this list more than `validatorLockedGoldRequirements.duration` seconds ago.

The reason we need to check to see if it's currently a member of the group with which it's affiliated is because the `lastRemovedFromGroupTimestamp` isn't relevant information if the validator is still a member of a group.